### ISSUE TITLE
Can't get right prediction when limiting the parameter search range in inverse problem

### ISSUE THREAD

**Issue by ZSTanone** [lululxvi/deepxde#644]
Hi ，Dr.Lu
After the program worked  without error, I started to limit the search for parameters by using `def get_variable` to limit the range of parameters which you mentioned in SBINN paper.
I use the Inverse problem for the Lorenz system as  example. Here I run the original code ，and the result shows I got the correct value of parameters. 
![image](https://user-images.githubusercontent.com/103397565/166106978-8e4a7400-3cc0-472b-85af-d56b149839f7.png)

However something wrong has occurred when I use `def get_variable`.

`    
    
    def get_variable(v, var):
        low, up = v * 0.2, v * 1.8
        l = (up - low) / 2
        v1 = l * torch.tanh(var) + l + low
        return v1
    C1_ = dde.Variable(1.0)
    C2_ = dde.Variable(1.0)
    C3_ = dde.Variable(1.0)
    
    
    def Lorenz_system(x, y):
        """Lorenz system.
        dy1/dx = 10 * (y2 - y1)
        dy2/dx = y1 * (15 - y3) - y2
        dy3/dx = y1 * y2 - 8/3 * y3
        """
        y1, y2, y3 = y[:, 0:1], y[:, 1:2], y[:, 2:]
    
        C1 = get_variable(12, C1_)  # 10
        C2 = get_variable(18, C2_)  # 15
        C3 = get_variable(3, C3_)   # 8/3
    
        dy1_x = dde.grad.jacobian(y, x, i=0)
        dy2_x = dde.grad.jacobian(y, x, i=1)
        dy3_x = dde.grad.jacobian(y, x, i=2)
        return [
            dy1_x - C1 * (y2 - y1),
            dy2_x - y1 * (C2 - y3) + y2,
            dy3_x - y1 * y2 + C3 * y3,
        ]
`
model.compile("adam", lr=0.001, external_trainable_variables=[C1_, C2_, C3_])`
`variable = dde.callbacks.VariableValue(
    [C1_, C2_, C3_], period=1000, filename="lorenz.dat"
)
`
After training about 30k，loss became small but the value of parameters incorrect
Loss：
![image](https://user-images.githubusercontent.com/103397565/166108444-ef279cdb-1dd7-49fe-b7fb-c89671893f77.png)

Parameters：
![image](https://user-images.githubusercontent.com/103397565/166108452-764d5503-dc37-4ae6-8b24-1c472eab1f66.png)

Could you tell me why this is happening? Thanks！！

**Comment by lululxvi** [comment]
What are the values of C1, C2, C3?

**Comment by ZSTanone** [comment]
> What are the values of C1, C2, C3?

Their true values are 10  15   8/3

And the nominal values I give are 12   18  3
 The predictions when using get variable func are
![image](https://user-images.githubusercontent.com/103397565/166228171-5e52e8b2-7179-4be1-b5c3-5445277b4a4d.png)

**Comment by lululxvi** [comment]
The figure you show is the values of C1_, etc.

**Comment by ZSTanone** [comment]
> The figure you show is the values of C1_, etc.
Here are C1 C2 C3
![image](https://user-images.githubusercontent.com/103397565/166631980-4dddf75e-e08e-45b5-9e31-6ae51cb97be4.png)

**Comment by lululxvi** [comment]
The results seem good.

**Comment by ZSTanone** [comment]
> The results seem good.

I think maybe I didn't express clearly , as you can see the target parameters are C1_、C2_、C3_ . I use get_variable to define C1,C2,C3, and give them a nominal value like 
![image](https://user-images.githubusercontent.com/103397565/167096060-258311ec-fc1d-4c50-8f04-dc8eac17368b.png)
The right answer ought be **10  15  8/3** respectively ，but the actual result is 
![image](https://user-images.githubusercontent.com/103397565/167096497-a7dfd815-b0f6-45a2-99b4-b7da3138dc87.png)
So I wonder what's wrong with this method（get_variable）.

**Comment by ZSTanone** [comment]
> > > The results seem good.
> > 
> > 
> > I think maybe I didn't express clearly , as you can see the target parameters are C1_、C2_、C3_ . I use get_variable to define C1,C2,C3, and give them a nominal value like ![image](https://user-images.githubusercontent.com/103397565/167096060-258311ec-fc1d-4c50-8f04-dc8eac17368b.png) The right answer ought be **10 15 8/3** respectively ，but the actual result is ![image](https://user-images.githubusercontent.com/103397565/167096497-a7dfd815-b0f6-45a2-99b4-b7da3138dc87.png) So I wonder what's wrong with this method（get_variable）.
> 
> You just misunderstand `C1` and `C1_`. The output of the model is `C1_`, but not `C1.` You need to use `get_variable` to the predicted value so that `C1_` can be converted to `C1`, like `C1=get_variable(12, C1_)`. In you case, `C1_` is -0.212, `low` and `up` are 2.4 and 21.6, `l` is 9.6. Thus `C1=9.6*tanh(-0.212)+9.6+2.4=9.99` Is it clear?

Much appreciate for your reply

**Comment by ZSTanone** [comment]
> > > The results seem good.
> > 
> > 
> > I think maybe I didn't express clearly , as you can see the target parameters are C1_、C2_、C3_ . I use get_variable to define C1,C2,C3, and give them a nominal value like ![image](https://user-images.githubusercontent.com/103397565/167096060-258311ec-fc1d-4c50-8f04-dc8eac17368b.png) The right answer ought be **10 15 8/3** respectively ，but the actual result is ![image](https://user-images.githubusercontent.com/103397565/167096497-a7dfd815-b0f6-45a2-99b4-b7da3138dc87.png) So I wonder what's wrong with this method（get_variable）.
> 
> You just misunderstand `C1` and `C1_`. The output of the model is `C1_`, but not `C1.` You need to use `get_variable` to the predicted value so that `C1_` can be converted to `C1`, like `C1=get_variable(12, C1_)`. In you case, `C1_` is -0.212, `low` and `up` are 2.4 and 21.6, `l` is 9.6. Thus `C1=9.6*tanh(-0.212)+9.6+2.4=9.99` Is it clear?
Thank you very much ，your reply helps me a lot！

**Comment by ZSTanone** [comment]
> > > The results seem good.
> > 
> > 
> > I think maybe I didn't express clearly , as you can see the target parameters are C1_、C2_、C3_ . I use get_variable to define C1,C2,C3, and give them a nominal value like ![image](https://user-images.githubusercontent.com/103397565/167096060-258311ec-fc1d-4c50-8f04-dc8eac17368b.png) The right answer ought be **10 15 8/3** respectively ，but the actual result is ![image](https://user-images.githubusercontent.com/103397565/167096497-a7dfd815-b0f6-45a2-99b4-b7da3138dc87.png) So I wonder what's wrong with this method（get_variable）.
> 
> You just misunderstand `C1` and `C1_`. The output of the model is `C1_`, but not `C1.` You need to use `get_variable` to the predicted value so that `C1_` can be converted to `C1`, like `C1=get_variable(12, C1_)`. In you case, `C1_` is -0.212, `low` and `up` are 2.4 and 21.6, `l` is 9.6. Thus `C1=9.6*tanh(-0.212)+9.6+2.4=9.99` Is it clear?
Thank you very much ，your reply helps me a lot！

**Comment by ZSTanone** [comment]
Thank you very much ，your reply helps me a lot！ @forxltk
